allennlp/nn/decoding/maximum_marginal_likelihood.py

```diff
@@ -81,23 +89,49 @@ def _create_allowed_transitions(targets: torch.Tensor,
 
         We use this to prune the set of actions we consider during decoding, because we only need
         to score the sequences in ``targets``.
-
-        While the example and types above say that this should be a 2D tensor, the implementation
-        logic is actually more flexible - any sequence of sequences is valid here, with any
-        contained types.  So your targets could be strings, for instance, if your ``DecoderStep``
-        and ``DecoderState`` objects know how to handle that.
         """
-        allowed_transitions: Dict[Tuple[int, ...], Set[int]] = defaultdict(set)
+        assert targets.dim() == 3, "targets tensor needs to be batched!"
```

Comment:
raise a `ConfigurationError`, rather than asserting

Reply:
The original idea behind `ConfigurationError` was that it was a single error that covered all kinds of problems in the config file specification, and could be caught and handled appropriately in some situations.  If you run into this error, it is a problem with your code, not with your configuration file, so `ConfigurationError` doesn't fit.